packages/hash/task-executor/src/airbyte/executor.ts

```diff
@@ -0,0 +1,167 @@
+/** @todo - Don't ignore Airbyte Log messages */
+
+export class BaseExecutor implements AirbyteExecutor {
+  imageName;
+
+  constructor(imageName: string) {
+    this.imageName = imageName;
+  }
+
+  async runSpec(): Promise<ConnectorSpecification> {
+    const response = await executeTask("docker", [
+      "run",
+      "--rm",
+      this.imageName,
+      "spec",
+    ]);
+
+    const messages = parseMessageStream(response);
+
+    const specMessage = messages.find(
+      (message) => message.type === "SPEC" && message.spec,
+    );
+    if (specMessage) {
+      return specMessage.spec!;
+    } else {
+      throw new Error("Message didn't contain a ConnectorSpecification");
+    }
+  }
+
+  async runCheck(config: any): Promise<AirbyteConnectionStatus> {
+    const configPath = await writeToTempFile(
+      CONFIG_FILE_NAME,
+      JSON.stringify(config),
+    );
+
+    const response = await executeTask("docker", [
+      "run",
+      "--rm",
+      "-v",
+      `${configPath}:/secrets/config.json`,
+      this.imageName,
+      "check",
+      "--config",
+      "/secrets/config.json",
+    ]);
+
+    const messages = parseMessageStream(response);
+
+    const statusMessage = messages.find(
+      (message) =>
+        message.type === "CONNECTION_STATUS" && message.connectionStatus,
+    );
+    if (statusMessage) {
+      return statusMessage.connectionStatus!;
+    } else {
+      throw new Error("Message didn't contain a AirbyteConnectionStatus");
+    }
+  }
+
+  async runDiscover(config: any): Promise<AirbyteStream[]> {
+    const configPath = await writeToTempFile(
+      CONFIG_FILE_NAME,
+      JSON.stringify(config),
+    );
+
+    const response = await executeTask("docker", [
+      "run",
+      "--rm",
+      "-v",
+      `${configPath}:/secrets/config.json`,
+      this.imageName,
+      "discover",
+      "--config",
+      "/secrets/config.json",
+    ]);
+
+    const messages = parseMessageStream(response);
+
+    const catalogMessage = messages.find(
+      (message) => message.type === "CATALOG" && message.catalog,
+    );
+    if (catalogMessage) {
+      return catalogMessage.catalog!.streams;
+    } else {
+      throw new Error("Message didn't contain a Airbyte Catalog");
+    }
+  }
+
+  async runRead(
+    config: any,
+    configuredCatalog: ConfiguredAirbyteCatalog,
+    statePath?: string,
+  ): Promise<RecordMessage[]> {
+    const configPath = await writeToTempFile(
+      CONFIG_FILE_NAME,
+      JSON.stringify(config),
+    );
+    const catalogPath = await writeToTempFile(
+      CATALOG_FILE_NAME,
+      JSON.stringify(configuredCatalog),
+    );
+
+    const args = [
+      "run",
+      "--rm",
+      "-v",
+      `${configPath}:/secrets/config.json`,
+      "-v",
+      `${catalogPath}:/secrets/catalog.json`,
+      this.imageName,
+      "read",
+      "--config",
+      "/secrets/config.json",
+      "--catalog",
+      "/secrets/catalog.json",
```

Comment:
These are the mounted volumes within the docker container. The `config.json` should be made in a temporary directory (see `writeToTempFile`) and shouldn't be located within the repo so adding it to `.gitignore` shouldn't do anything.

If that's not the case it's a bug or oversight.

Reply:
Gotcha, apologies. I'd like to make a .gitignore in the `packages/hash/task-executor` folder because when this is deleted it'll be easy to make sure the ignore record is deleted too. 

I think we have some discussion about trying to keep things in a global .gitignore though (which would be a lot easier to miss) @kachkaev do you see problems with me making ``packages/hash/task-executor/.gitignore` for `catalog.json`?

Reply:
I personally don't have an issue with having scoped `.gitignore` files for packages like this, but will defer to @kachkaev

Reply:
I don't mind having multiple `.gitignore` files, but from my experience this brings more complexities than benefits in a monorepo of our size. The cost of not removing some path from the root `.gitignore` is not that big compared to managing a fractal tree of configs. Deciding where to place a new file and what to put there is an extra human task that can be avoided. A single `.gitconfig` shows all the applied rules at a glance.

Up to you to decide. If we go for a new `.gitignore` file, let's chat about the rules of these files after this PR. We should have some sort of a system that helps us create the right number of `.gitignore` files with the right contents. Ideally, these rules should be simple and unambiguous. What I mean is that the number and the contents of `.gitignore` files should be a function of the today's state of the repo rather than its history.

Reply:
Makes sense, I thought there was a problem with the splitting the config due to some tooling not being able to resolve the child directories but maybe I'm misremembering and you were talking about something other than .gitignore.

I think for now I'll go ahead with adding one in this folder purely because we do plan on fully deleting the `task-executor` package and it's purposefully pretty self-contained.

Lets have a follow-up discussion as you say.